packages/x-charts/src/ChartsLegend/ChartsLegendItem.tsx

```diff
@@ -0,0 +1,87 @@
+import * as React from 'react';
+import clsx from 'clsx';
+import { useRtl } from '@mui/system/RtlProvider';
+import { ChartsText, ChartsTextStyle } from '../ChartsText';
+import { LegendItemParams } from './chartsLegend.types';
+import { ChartsLegendClasses } from './chartsLegendClasses';
+
+export interface ChartsLegendItemProps extends LegendItemParams {
+  positionY: number;
+  label: string;
+  positionX: number;
+  innerHeight: number;
+  innerWidth: number;
+  color: string;
+  gapX: number;
+  gapY: number;
+  legendWidth: number;
+  itemMarkHeight: number;
+  itemMarkWidth: number;
+  markGap: number;
+  labelStyle: ChartsTextStyle;
+  classes?: Omit<Partial<ChartsLegendClasses>, 'column' | 'row' | 'label'>;
+  onClick?: (event: React.MouseEvent<SVGRectElement, MouseEvent>) => void;
+}
+
+/**
+ * @ignore - internal component.
+ */
+function ChartsLegendItem(props: ChartsLegendItemProps) {
+  const isRTL = useRtl();
+  const {
+    id,
+    positionY,
+    label,
+    positionX,
+    innerHeight,
+    innerWidth,
+    legendWidth,
+    color,
+    gapX,
+    gapY,
+    itemMarkHeight,
+    itemMarkWidth,
+    markGap,
+    labelStyle,
+    classes,
+    onClick,
+  } = props;
+
+  return (
+    <g
+      className={clsx(classes?.series, `${classes?.series}-${id}`)}
+      transform={`translate(${gapX + (isRTL ? legendWidth - positionX : positionX)} ${gapY + positionY})`}
+    >
+      <rect
+        x={isRTL ? -(innerWidth + 2) : -2}
+        y={-itemMarkHeight / 2 - 2}
+        width={innerWidth + 4}
+        height={innerHeight + 4}
```

Comment:
Why does the background rectangle has an extra 2px on every direction ?

Reply:
I added it because it looks better when styling the hover, just like a button usually has a margin.

And since it is difficult to adjust the svg I thought it was useful, though I'm open to suggestions 😄

Reply:
It's ok, just wanted to be sure it's not experimentation that got committed and forgotten ;)